### PR TITLE
Add warnings to stitchers and options to merge boundaries.

### PR DESCRIPTION
**Pull request by grmnptr**
Refs #28049

### CODE REVIEW

File: framework/src/utils/MooseMeshUtils.C [hunk 1]
```diff
@@ -21,6 +21,25 @@ using namespace libMesh;
...
+  // first ID to every occurrence.
+  const auto & bd_name_map = mesh.get_boundary_info().get_sideset_name_map();
+  std::map<boundary_id_type, boundary_id_type> same_name_ids;
+  for (const auto & pair_outer : bd_name_map)
+    for (const auto & pair_inner : bd_name_map)
+      // The last condition is needed to make sure we only store one combination
+      if (pair_outer.second == pair_inner.second && pair_outer.first != pair_inner.first &&
+          same_name_ids.find(pair_inner.first) == same_name_ids.end())
+        same_name_ids[pair_outer.first] = pair_inner.first;
+
```
Comment (GiudGiud) [inline framework/src/utils/MooseMeshUtils.C:38]: I think you would want to duplicate the code above but for the nodeset_name_map

File: framework/src/meshgenerators/MeshRepairGenerator.C [hunk 2]
```diff
@@ -37,6 +37,10 @@ MeshRepairGenerator::validParams()
                         false,
                         "Create new blocks if multiple element types are present in a block");
 
+  params.addParam<bool>("merge_boundary_ids_with_same_name",
+                        false,
```
Comment (roystgnr) [inline framework/src/meshgenerators/MeshRepairGenerator.C:41]: Default to false ... but when do we *not* want to do this?  We treat names as first-class identifiers in MOOSE, and we *only* treat numeric ids as first-class identifiers in libMesh, so if we ever have a single name matching multiple ids isn't that just a bug waiting to happen?

Comment (grmnptr) [reply framework/src/meshgenerators/MeshRepairGenerator.C:41]: The reason behind this is that when I looked at the object it looked like every other manipulation defaulted to false. It feels like to control what this guy does you have to manually set the operation to true. So I just went with the current design.

**Comment by GiudGiud** [comment]
missing an include
https://civet.inl.gov/job/2334684/

**Comment by GiudGiud** [comment]
this is going to break some meshes for us on the VTB but I ll know what to do

File: framework/src/meshgenerators/MeshRepairGenerator.C (hunk 2 above)
Comment (roystgnr) [reply framework/src/meshgenerators/MeshRepairGenerator.C:41]: Ah, that's right; this is the `MeshRepairGenerator`, we're trying to be as non-intrusive as possible here.  Question rescinded.

**Comment by GiudGiud** [comment]
one last failure there

**Comment by grmnptr** [comment]
I can prepare the VTB PR too if you want me to.

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**